Adds Open Graph tags

gamebits opened the pull request:
Adds `og` tags for social media. Assumes that this service does not have its own Twitter or Facebook profiles, or else additional tags will be needed.

Also uploads favicon images, though I note `index.html` already points to seemingly non-existent such files in the root directory.

Commit:
Adds Open Graph tags

Adds `og` tags for social media. Assumes that this service does not have its own Twitter or Facebook profiles, or else additional tags will be needed.

Also uploads favicon images, though I note `index.html` already points to seemingly non-existent such files in the root directory.

Commit:
Merge branch 'main' of https://github.com/gamebits/dealwithit

klimeryk commented on index.html:
```diff
@@ -9,6 +9,20 @@
+  <meta property="og:description" content="All done artisanally and securely in your browser." />
+  <meta property="og:url" content="https://emoji.build/deal-with-it-generator/" />
+  <meta property="og:site_name" content="Deal With It GIF emoji generator" />
+  <meta property="og:image" content="https://emoji.build/deal-with-it-generator/favicon.ico" />
```
```suggestion
  <meta property="og:image" content="/favicon.ico" />
```

There's favicons in https://github.com/klimeryk/dealwithit/tree/main/public - could you please use them instead? 🙏

gamebits replied:
i.e., I should be using `https://emoji.build/favicon.ico` — is that right? I get a 404 on that URL:

```
$ wget https://emoji.build/favicon.ico
--2024-10-15 11:37:09--  https://emoji.build/favicon.ico
Resolving emoji.build (emoji.build)... 172.67.160.63, 104.21.9.152
Connecting to emoji.build (emoji.build)|172.67.160.63|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2024-10-15 11:37:09 ERROR 404: Not Found.
```

klimeryk replied:
No, just use `/favicon.ico`. Don't include the full domain URL. That will make it tied to that domain (which might not be what you want, for example, for local development or if deploying to a different domain for testing purposes). Normally, `/favicon.ico` would mean indeed mean `https://emoji.build/favicon.ico`. But Vite (the build system I'm using) automatically corrects the path at compile time: https://vite.dev/guide/build.html#public-base-path. So it ends up `https://emoji.build/deal-with-it-generator/favicon.ico`

klimeryk commented on index.html:
```diff
@@ -9,6 +9,20 @@
+  <meta name="twitter:card" content="summary" />
+  <meta name="twitter:domain" content="emoji.build" />
+  <meta name="twitter:title" content="Deal With It GIF emoji generator" />
+  <meta name="twitter:description" content="All done artisanally and securely in your browser." />
```
```suggestion
  <meta name="twitter:description" content="Most over-engineered Deal With It emoji generator - all done securely in your browser for free." />
```

Something like this might be a better description.

gamebits replied:
Sure thing! I wasn't sure to what degree I wanted to exercise editorial creativity over someone else's site, so wherever possible, I copy/pasted existing assets. Happy to use this instead!

klimeryk reviewed (commented):
Thanks for the suggestion! I missed this metadata. And you're correct that this does not have a Facebook or Twitter page (nor do I 😅 ).
I've left some feedback 🙇

Commit:
Updates Twitter description

Co-authored-by: Igor Klimer <klimeryk@users.noreply.github.com>

Commit:
Updates favicons

Confirmed with @klimeryk that favicons already existed. Paths have been updated!

Commit:
Update index.html

Commit:
Merge branch 'main' of https://github.com/gamebits/dealwithit

gamebits commented:
@klimeryk Do you want `og:description` ("All done artisanally and securely in your browser") updated to match the revised `twitter:description` ("Most over-engineered Deal With It emoji generator - all done securely in your browser for free")?

klimeryk commented:
Sorry for the (_extremely_) late follow-up here, @gamebits 🙇‍♂️ I went ahead and updated the `og:description` to match `twitter:description` (good catch!), rebased, squashed and pushed to main in https://github.com/klimeryk/dealwithit/commit/2cad259fcb313bbd692d022f7581074185df3d00 (while making sure to preserve you as the author of the change). Thank you!
